packages/zwave-js/src/lib/serialapi/capability/LongRangeSetupMessages.ts

```diff
@@ -0,0 +1,169 @@
+import { MessagePriority, encodeBitMask, parseBitMask } from "@zwave-js/core";
+import type { ZWaveHost } from "@zwave-js/host";
+import type { SuccessIndicator } from "@zwave-js/serial";
+import {
+	FunctionType,
+	Message,
+	type MessageBaseOptions,
+	type MessageDeserializationOptions,
+	MessageType,
+	expectedResponse,
+	gotDeserializationOptions,
+	messageTypes,
+	priority,
+} from "@zwave-js/serial";
+
+// TODO: move to header?
+export enum LongRangeChannel {
+	Unknown = 0x00, // Reserved
+	A = 0x01,
+	B = 0x02,
+	// 0x03..0xFF are reserved and must not be used
+}
+
+@messageTypes(MessageType.Request, FunctionType.GetLongRangeChannel)
+@expectedResponse(FunctionType.GetLongRangeChannel)
+@priority(MessagePriority.Controller)
+export class GetLongRangeChannelRequest extends Message {}
+
+export interface LongRangeChannelResponseOptions extends MessageBaseOptions {
+	longRangeChannel: LongRangeChannel;
+}
+
+export class LongRangeChannelMessageBase extends Message {
+	public constructor(
+		host: ZWaveHost,
+		options:
+			| MessageDeserializationOptions
+			| LongRangeChannelResponseOptions,
+	) {
+		super(host, options);
+
+		if (gotDeserializationOptions(options)) {
+			// BUGBUG: verify length?
+			switch (this.payload[0]) {
+				case 0x01:
+					this.longRangeChannel = LongRangeChannel.A;
+					break;
+				case 0x02:
+					this.longRangeChannel = LongRangeChannel.B;
+					break;
+				default:
+					this.longRangeChannel = LongRangeChannel.Unknown;
+					break;
+			}
+		} else {
+			this.longRangeChannel = options.longRangeChannel;
+		}
+	}
+
+	public longRangeChannel: LongRangeChannel;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(1);
+
+		switch (this.longRangeChannel) {
+			default: // Don't use reserved values, default back to A
+			case LongRangeChannel.A:
+				this.payload[0] = LongRangeChannel.A;
+				break;
+
+			case LongRangeChannel.B:
+				this.payload[0] = LongRangeChannel.B;
+		}
+
+		return super.serialize();
+	}
+}
+
+@messageTypes(MessageType.Response, FunctionType.GetLongRangeChannel)
+@priority(MessagePriority.Controller)
+export class GetLongRangeChannelResponse extends LongRangeChannelMessageBase {}
+
+@messageTypes(MessageType.Request, FunctionType.SetLongRangeChannel)
+@expectedResponse(FunctionType.SetLongRangeChannel)
+@priority(MessagePriority.Controller)
+export class SetLongRangeChannelRequest extends LongRangeChannelMessageBase {
+}
+
+export interface SetLongRangeChannelResponseOptions extends MessageBaseOptions {
+	responseStatus: number;
+}
+
+// BUGBUG:  move someplace common, see Spec 4.2.5
+//
+// Although, that section refers to a "callback message", and SetZWaveLongRangeChannel doesn't define a callback message...
+export enum ResponseStatus {
+	FAILED = 0x00,
+}
```

Comment:
The `ReportStatus` here is supposed to be a common thing, but in looking through other things that use it, I didn't see one that was implemented. It's _supposed_ to be used with messages that then have a callback message, so that a failure can stop the callback from occuring, but this message doesn't have a callback message...

Seems like a case where this should have just been pass/fail on it's own, and someone in the spec committee decided to re-use a status field that wasn't necessarily applicable?

Reply:
> It's _supposed_ to be used with messages that then have a callback message, so that a failure can stop the callback from occuring, but this message doesn't have a callback message...

In that case the response indicates the final result of the command flow, not the callback.

I don't think we need the enum value or a separate field. Just do this like we do here for example:
https://github.com/jtbraun/node-zwave-js/blob/d1e23fc6a2e58c15eff86d31cf78951d8b62476b/packages/zwave-js/src/lib/serialapi/capability/SerialAPISetupMessages.ts#L899-L907